### PR TITLE
Cr 468 update headers

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Add security headers to RHUI

# What has changed

- Adds security CSP headers to pages

- Update to Design System version to bring through revised nonce implementation

- Added passing of nonce is pageConfig, and pick up in inline javascript calls

- Update of /start breadcrumb as needed by Design System update

- Fix of welsh accessibility link (other changes made in this template)

**YOU WILL NEED TO TRIGGER AN UPDATE OF THE DESIGN SYSTEM TO TEST**

# How to test?
Run site, and check via the inspector there are no page errors relating to security and javascript
**This needs testing against EQ and later campaign site, but needs to be in whitelodge for this**

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-468

### CODE REVIEW

**Comment by AmeeshKutthan** [comment]
Will review as a priority

**Review by cookel2 — COMMENTED** [review]
just one comment there

File: app/security.py [hunk 1]
```diff
@@ -8,27 +8,66 @@
...
-            'https://cdn.ons.gov.uk'
-        ],
-        'img-src': [
-            "'self'", 'data:', 'https://www.google-analytics.com',
-            'https://cdn.ons.gov.uk'
-        ],
-    },
-    'X-XSS-Protection': '1',
+    'Strict-Transport-Security': 'max-age=31536000 includeSubDomains',
+    # 'Content-Security-Policy': _format_csp(CSP),
```
Comment (AmeeshKutthan) [inline app/security.py:64]: perhaps delete this comment

**Review by AmeeshKutthan — APPROVED** [review]
Code Review:

The following is a code review for CR-468.
I have utilised the following reference documentation, when reviewing this CR:

https://www.bounteous.com/insights/2017/07/20/using-google-analytics-and-google-tag-manager-content-security-policy/

https://developers.google.com/web/fundamentals/security/csp

https://owasp.org/www-community/xss-filter-evasion-cheatsheet

Below is an example of the expected Security Header response.

**Expected Security Header:**

alt-svc: clear

cache-control: no-cache, no-store, must-revalidate

content-length: 247

content-security-policy: default-src 'self' https://cdn.eq.census-gcp.onsdigital.uk; font-src 'self' data: https://fonts.gstatic.com https://cdn.eq.census-gcp.onsdigital.uk; script-src 'self' https://www.googletagmanager.com 'unsafe-inline' 'unsafe-eval' https://cdn.eq.census-gcp.onsdigital.uk 'nonce-GDXhGo7zV6Drj_friKk1cw'; style-src 'self' https://tagmanager.google.com https://fonts.googleapis.com 'unsafe-inline' https://cdn.eq.census-gcp.onsdigital.uk; connect-src 'self' https://cdn.eq.census-gcp.onsdigital.uk https://whitelodge-ai-api.census-gcp.onsdigital.uk; frame-src https://www.googletagmanager.com; img-src 'self' data: https://www.google-analytics.com https://ssl.gstatic.com https://www.gstatic.com https://cdn.eq.census-gcp.onsdigital.uk

content-type: text/html; charset=utf-8

date: Fri, 25 Sep 2020 09:54:57 GMT

location: https://test-runner.gcp.dev.eq.ons.digital/questionnaire/name/

pragma: no-cache

referrer-policy: strict-origin-when-cross-origin

server: None

set-cookie: session=<COOKIE_DATA>; Secure; HttpOnly; Path=/; SameSite=Strict

status: 302

strict-transport-security: max-age=31536000; includeSubDomains

vary: Cookie

via: 1.1 google

x-content-security-policy: default-src 'self' https://cdn.eq.census-gcp.onsdigital.uk; font-src 'self' data: https://fonts.gstatic.com https://cdn.eq.census-gcp.onsdigital.uk; script-src 'self' https://www.googletagmanager.com 'unsafe-inline' 'unsafe-eval' https://cdn.eq.census-gcp.onsdigital.uk 'nonce-GDXhGo7zV6Drj_friKk1cw'; style-src 'self' https://tagmanager.google.com https://fonts.googleapis.com 'unsafe-inline' https://cdn.eq.census-gcp.onsdigital.uk; connect-src 'self' https://cdn.eq.census-gcp.onsdigital.uk https://whitelodge-ai-api.census-gcp.onsdigital.uk; frame-src https://www.googletagmanager.com; img-src 'self' data: https://www.google-analytics.com https://ssl.gstatic.com https://www.gstatic.com https://cdn.eq.census-gcp.onsdigital.uk

x-content-type-options: nosniff

x-frame-options: DENY

x-xss-protection: 1; mode=block

Below is an example of the Actual response Security Header, which, bar URL's and timestamps etc., is more or less the same, thus being a valid 
response:

**Actual Security Header:**

alt-svc: clear

content-length: 46292

content-security-policy: default-src 'self' https://cdn.ons.gov.uk; font-src 'self' data: https://fonts.gstatic.com https://cdn.ons.gov.uk;script-src 'self' https://www.googletagmanager.com 'unsafe-inline' 'unsafe-eval' https://cdn.ons.gov.uk 'nonce-SNHcQjvDuF4xbp3N'; style-src 'self' https://tagmanager.google.com https://fonts.googleapis.com 'unsafe-inline' https://cdn.ons.gov.uk; connect-src 'self' https://cdn.ons.gov.uk; frame-src https://www.googletagmanager.com https://www.timeforstorm.com; img-src 'self' data: https://www.google-analytics.com https://ssl.gstatic.com https://www.gstatic.com https://cdn.ons.gov.uk

content-type: text/html; charset=utf-8

date: Sun, 25 Oct 2020 16:04:24 GMT

referrer-policy: strict-origin-when-cross-origin

server: Python/3.6 aiohttp/3.6.2

status: 200

strict-transport-security: max-age=31536000 includeSubDomains

via: 1.1 google

x-content-security-policy: default-src 'self' https://cdn.ons.gov.uk; font-src 'self' data: https://fonts.gstatic.com https://cdn.ons.gov.uk; script-src 'self'https://www.googletagmanager.com 'unsafe-inline' 'unsafe-eval' https://cdn.ons.gov.uk 'nonce-SNHcQjvDuF4xbp3N'; style-src 'self' https://tagmanager.google.com https://fonts.googleapis.com 'unsafe-inline' https://cdn.ons.gov.uk; connect-src 'self' https://cdn.ons.gov.uk; frame-src https://www.googletagmanager.com https://www.timeforstorm.com; img-src 'self' data: https://www.google-analytics.com https://ssl.gstatic.com https://www.gstatic.com https://cdn.ons.gov.uk

x-content-type-options: nosniff

x-frame-options: DENY

x-xss-protection: 1; mode=block

As one can observe, they are almost the same. All the required 'directives' have been implemented correctly.
The newly implemented 'nonce' values are also valid, unique (per page request) and secure.

As such, all provided tests PASSED successfully.

Various screenshots detailed below:

- Valid response, EN site version, various examples across site:

Screenshot 2020-10-25 at 15 10 44

Screenshot 2020-10-25 at 15 12 15

Screenshot 2020-10-25 at 15 11 34

Screenshot 2020-10-25 at 15 11 06

Screenshot 2020-10-25 at 15 19 36

Screenshot 2020-10-25 at 15 15 03

Screenshot 2020-10-25 at 15 19 15
Screenshot 2020-10-25 at 15 18 35

Screenshot 2020-10-25 at 15 16 40

Screenshot 2020-10-25 at 15 16 23

Screenshot 2020-10-25 at 15 16 03

Screenshot 2020-10-25 at 15 15 43

- Valid response, CY site version:

Screenshot 2020-10-25 at 15 26 24

- Valid response, NI site version:

Screenshot 2020-10-25 at 15 28 31

- Valid response, EN Webchat site version:

Screenshot 2020-10-25 at 15 14 12

- Valid response, EN site version, error message examples:

Screenshot 2020-10-25 at 15 20 56

Screenshot 2020-10-25 at 15 55 23

Screenshot 2020-10-25 at 16 03 03

Screenshot 2020-10-25 at 16 01 55

Screenshot 2020-10-25 at 15 55 23

Screenshot 2020-10-25 at 15 54 12

Screenshot 2020-10-25 at 15 28 31

Screenshot 2020-10-25 at 15 34 07

Screenshot 2020-10-25 at 15 32 52

- Chrome Security validation screen:

Screenshot 2020-10-25 at 15 23 19

File: app/templates/base-cy.html [hunk 2]
```diff
@@ -83,7 +83,7 @@
                     },
                     {
                         'text': 'Hygyrchedd',
-                        'url': domain_url_en + '/cy/start/accessibility/'
+                        'url': domain_url_en + '/hygyrchedd/'
```
[the remark this answers is not in the document]
Comment (Paul-Joel) [reply app/templates/base-cy.html]: Well spotted - change made

**Review by cookel2 — APPROVED** [review]

**Pull request merged**